pkg/controllers/consolidation/controller.go

```diff
@@ -685,3 +630,94 @@ func (c *Controller) setNodeUnschedulable(ctx context.Context, nodeName string,
 	}
 	return nil
 }
+
+// validateCommand is used to validate that a lifecycleCommand is still able to be performed by re-inspecting nodes and
+// re-simulating scheduling
+// nolint:gocyclo
```

Comment:
I'm noticing a lot of disabling of gocyclo. We originally added it as a hint to developers to factor their code into smaller chunks to improve readability. 

I'm open minded about this though. Are you finding it getting in your way? Do we want to remove it from the project?

Reply:
I'll revisit and see if I can refactor. In general I think it's pretty useful, it's sort of a judgement call to see if breaking it up makes it worse or not.  I'll mark this as WIP, working on figuring out why it's failing 100% in CI and passing 100% locally.